src/averageAges.js

```diff
@@ -54,8 +87,31 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenWithChildren = onlyWithSon
+    ? people.filter(person =>
+      people.find(child =>
+        findMother(child, person)
+        && findMale(child)))
```

Comment:
can we increase the indentation, nested functions are hard to read. I kinda remember Paul telling 'never use nested arrow functions' but I am not sure. Maybe mentor's will have a comment about that.